test/generators/csharp/presets/__snapshots__/JsonSerializerPreset.spec.ts.snap

```diff
@@ -79,24 +79,24 @@ internal class TestConverter : JsonConverter<Test>
         }
       if (propertyName == \\"numberProp\\")
         {
-          var value = JsonSerializer.Deserialize<double>(ref reader, options);
+          var value = JsonSerializer.Deserialize<double?>(ref reader, options);
           instance.numberProp = value;
           continue;
         }
       if (propertyName == \\"enumProp\\")
         {
-          var value = EnumTestExtension.ToEnumTest(JsonSerializer.Deserialize<dynamic>(ref reader, options));
+          var value = EnumTest?Extension.ToEnumTest?(JsonSerializer.Deserialize<dynamic>(ref reader, options));
```

Comment:
This is definitely wrong 😆 

You need to change this line: https://github.com/asyncapi/modelina/blob/3cfd68b877d04f48c492b1eb53cf58b1c2d7179d/src/generators/csharp/presets/JsonSerializerPreset.ts#L91

to use the `.name` instead of type.